src/Storages/FileLog/StorageFileLog.cpp

```diff
@@ -95,8 +95,9 @@ StorageFileLog::StorageFileLog(
 void StorageFileLog::loadMetaFiles(bool attach)
 {
     const auto & storage = getStorageID();
+    /// FIXME Why do we need separate directory? Why not to use data directory?
     root_meta_path
-        = std::filesystem::path(getContext()->getPath()) / ".filelog_storage_metadata" / storage.getDatabaseName() / storage.getTableName();
+        = std::filesystem::path(getContext()->getPath()) / "filelog_storage_metadata" / DatabaseCatalog::getPathForUUID(storage.uuid);
```

Comment:
```suggestion
        = std::filesystem::path(getContext()->getPath()) / "stream_engines/filelog/" / DatabaseCatalog::getPathForUUID(storage.uuid);
```
(https://clickhouse-inc.slack.com/archives/C02F2LML5UG/p1644408501757029?thread_ts=1644372645.666099&cid=C02F2LML5UG)

Reply:
Btw, why not to use data directory as all table engines do?

Reply:
I am not sure, but I remember it was attempted and there was some problem with unexpected file exception because only certain files were expected in that directory (but possibly it was in a different place, I do not remember for sure now).

Reply:
> unexpected file exception because only certain files were expected in that directory

But table engine fully controls content of its data directory, so probably it was a bug in `StorageFileLog`. Let's try again in another PR.

> there are other engines which can create some unrelated to general data directory files?

AFAIK no